File: frontend/fitpassapp/src/app/models/Manages.ts
export class Manages {
    
    id?: number;
    endDate?: Date;
    startDate?: Date;
    facilityId: number;
    userId: number;

    constructor(userId: number, facilityId: number,startDate?: Date, endDate?: Date,id?: number) {
        this.id = id;
        this.endDate = endDate;
        this.startDate = startDate;
        this.facilityId = facilityId;
        this.userId = userId;
    }

}

File: backend/fitpass/src/main/java/fitpass/fitpass/model/entity/Manages.java
package fitpass.fitpass.model.entity;


import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.NoArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "manages")
public class Manages {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalDate endDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id" , referencedColumnName = "id")
    @JsonBackReference
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "facility_id" , referencedColumnName = "id")
    @JsonBackReference
    private Facility facility;

//    @Column(name = "is_deleted",nullable = false)
//    private Boolean isDeleted = false;
}
